x/programs/rust/examples/single_nft/src/example.rs

```diff
@@ -0,0 +1,139 @@
+                    param_type: ParamType::U64,
+                    value: nft_name_length,
+                },
+                Param {
+                    param_type: ParamType::String,
+                    value: nft_symbol,
+                },
+                Param {
+                    param_type: ParamType::U64,
+                    value: nft_symbol_length,
+                },
+                Param {
+                    param_type: ParamType::String,
+                    value: nft_uri,
+                },
+                Param {
+                    param_type: ParamType::U64,
+                    value: nft_uri_length,
+                },
+            ],
+            require: None,
+        },
+        Step {
+            endpoint: Endpoint::Key,
+            method: "key_create".to_string(),
+            max_units: 0,
+            params: vec![Param {
+                param_type: ParamType::Key(Key::Ed25519),
+                value: "alice_key".to_string(),
+            }],
+            require: None,
+        },
+        Step {
+            endpoint: Endpoint::Execute,
+            method: "mint".to_string(),
+            max_units: 100000,
+            params: vec![
+                Param {
+                    param_type: ParamType::Id,
+                    value: id_from_step(0),
+                },
+                Param {
+                    param_type: ParamType::Key(Key::Ed25519),
+                    value: "alice_key".to_string(),
+                },
+            ],
+            require: None,
+        },
+        Step {
+            endpoint: Endpoint::Execute,
+            method: "burn".to_string(),
+            max_units: 100000,
+            params: vec![
+                Param {
+                    param_type: ParamType::Id,
+                    value: id_from_step(0),
+                },
+                Param {
+                    param_type: ParamType::Key(Key::Ed25519),
+                    value: "alice_key".to_string(),
+                },
+            ],
+            require: None,
+        },
+    ];
+
+    Plan {
+        caller_key: "alice_key".to_string(),
+        steps,
+    }
+}
+
+#[allow(dead_code)]
+fn run() -> Result<Vec<PlanResponse>, Box<dyn std::error::Error>> {
+    let nft_name = "MyNFT".to_string();
+    let binding = nft_name.len().to_string();
+    let nft_name_length = binding.to_string();
+
+    let nft_symbol = "MNFT".to_string();
+    let binding = nft_symbol.len().to_string();
+    let nft_symbol_length = binding.to_string();
+
+    let nft_uri = "ipfs://my-nft.jpg".to_string();
+    let binding = nft_uri.len().to_string();
+    let nft_uri_length = binding.to_string();
+
+    let plan = initialize_plan(
+        nft_name,
+        nft_name_length,
+        nft_symbol,
+        nft_symbol_length,
+        nft_uri,
+        nft_uri_length,
+    );
+    let client = Client::new("path to simulator".to_owned());
+
+    let tx = client.run::<PlanResponse>(&plan)?;
```

Comment:
```suggestion
    let tx = client.run(&plan)?;
```

Reply:
Do you consider this blocking?

Reply:
Over-specifying code is a maintenance burden, and turbofish syntax is generally harder to read. Frameworks live and die by the simplicity of their examples, so yes, I would argue that this is blocking. 

Regardless, why wouldn't you apply this change?

Reply:
Because this way is more clear to me.

Reply:
Also, if I remove the generic, I get this error:
```
consider giving this closure parameter an explicit type, where the type for type parameter `T` is specified: `: &T`rustc[E0282](https://doc.rust-lang.org/error-index.html#E0282)
single_nft.rs(32, 20): original diagnostic
type annotations needed for `&T`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B2%5D?2#file%3A%2F%2F%2FUsers%2Fdan.sover%2Fcode%2Fhypersdk%2Fx%2Fprograms%2Frust%2Fexamples%2Fsingle_nft%2Fexamples%2Fsingle_nft.rs)
single_nft.rs(32, 26): type must be known at this point
single_nft.rs(32, 24): consider giving this closure parameter an explicit type, where the type for type parameter `T` is specified: `: &T`
```

Reply:
Yeah, the code change requires this now. I still suggest using the following where possible:
```rust
let plan_response: PlanResponse = client.run(*plan);
```
In general, this looks like a bit of a code-smell. It appears as though the `run` method was originally generic, but should now return a concrete type. Looking at the code, it also appears to be over-constrained.

In any case, specifying the type is now required

Reply:
cc @hexfusion  ^